pkg/apis/core/validation/validation.go

```diff
@@ -2957,6 +2957,18 @@ func ValidatePodSpec(spec *core.PodSpec, fldPath *field.Path) field.ErrorList {
 		for _, msg := range ValidateServiceAccountName(spec.ServiceAccountName, false) {
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("serviceAccountName"), spec.ServiceAccountName, msg))
 		}
+	} else {
```

Comment:
`ValidatePodSpec` is called from `ValidatePodTemplateSpec`, which is run on pod templates embedded in objects like replicasets and replicationcontrollers. `spec.serviceAccountName` can be empty in those when it would not be in the final pod. do we want to prevent a replicaset or deployment using a projected serviceAccountToken volume source without specifying an explicit service account name?

Reply:
if we want to do this only for actual Pod objects, move this to `ValidatePod`